server.js

```diff
@@ -1,30 +1,131 @@
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// START PAGE
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json({responseMessage: "Hello Dance Music!"});
+});
+
+// ALL TRACKS - works
+app.get("/tracks", (req, res) => {
+  res.status(200).json({topMusicData})
+});
+
+// ALL TRACKS IN REVERSED ORDER - works
+app.get("/tracks/reversed", (req, res) => {
+  const topMusicDataReversed = topMusicData.reverse()
+  res.status(200).json({topMusicDataReversed});
+});
+
+//TRACKS BY ID - works
+app.get("/tracks/:id", (request, response) => {
+  const singleTrack = topMusicData.find((track) => {
+    return track.id === Number(request.params.id);
+  });
+  if(singleTrack) {
+    response.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        tracks: singleTrack
+      }
+    });
+  } else {
+    response.status(404).json({
+      success: false,
+      message: "Not Found",
+      body: {}
+    });
+  }
+  console.log(singleTrack);
+});
+
+//TRACKS SEARCH BY BPM - works
+app.get("/tracks/bpm/:bpm", (request, response) => {
+  const singleTrack = topMusicData.find((track) => {
+    return track.bpm === Number(request.params.bpm);
+     });
+  if(singleTrack) {
+    response.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        tracks: singleTrack
+      }
+    });
+  } else {
+    response.status(404).json({
+      success: false,
+      message: "No tracks found with this BPM",
+      body: {}
+    });
+  }
+  console.log(singleTrack);
 });
 
 // Start the server
 app.listen(port, () => {
   console.log(`Server running on http://localhost:${port}`);
 });
+
+//OTHER TEST ROUTES WHICH DOES NOT WORK ATM ---------------------
+
+// RANDOM TRACK - does not work!
+app.get("/random-track", (req, res) => {
+  const topMusicDataRandom = topMusicData[Math.floor(Math.random()* data.length)]
+  if(!topMusicDataRandom) {
+    res.status(404).json({
+      data: "No random track found, please try again.",
+      success: false, 
+    })
+  } else {
+    res.status(200).json({
+      data: topMusicDataRandom,
+      success: true,
+    })
+  }
+})
+
+//DANCEABILITY : SORT FROM HIGHER TO LOWER RATE - does not work
+app.get("/tracks/danceability/", (request, response) => {
+  const singleTrack = topMusicData.find((track) => {
```

Comment:
Could it be that you use find() because if find() finds what it is looking for it will stop looking even if there are more finds available. If you understand what I mean.

Reply:
Nina had the solution so I guess you have already looke into that ;)